clustercontroller-core/src/main/java/com/yahoo/vespa/clustercontroller/core/NodeStateChangeChecker.java

```diff
@@ -281,48 +270,38 @@ private static boolean groupContainsNode(Group group, Node node) {
         return false;
     }
 
-    private Result anyNodeSetToMaintenance(ClusterState clusterState) {
-        for (NodeInfo nodeInfo : clusterInfo.getAllNodeInfo()) {
-            if (clusterState.getNodeState(nodeInfo.getNode()).getState() == State.MAINTENANCE) {
-                return Result.createDisallowed("Another node is already in maintenance:" + nodeInfo.getNodeIndex());
+    private Result checkAllNodesAreUp(ClusterState clusterState) {
+        // This method verifies both storage nodes and distributors are up (or retired).
+        // The complicated part is making a summary error message.
+
+        for (NodeInfo storageNodeInfo : clusterInfo.getStorageNodeInfo()) {
+            State wantedState = storageNodeInfo.getUserWantedState().getState();
+            if (wantedState != State.UP && wantedState != State.RETIRED) {
+                return Result.createDisallowed("Another storage node wants state " +
+                        wantedState.toString().toUpperCase() + ": " + storageNodeInfo.getNodeIndex());
```

Comment:
Is it possible for this to cause upgrade "starvation" by nodes tagged as `permanently down`? Or are such nodes kept around for such a brief period of time that it does not matter in practice?

Reply:
When a node is permanently set down, the application is redeployed immediately to remove the node.  If such a redeployment fails, e.g. a redeployment requires a new node but there is no spare capacity, then the permanently down node will linger and starve upgrades++.  I haven't witnessed this yet.

Come to think of it -- requiring all other nodes are up when setting a node permanently down seems too stringent.  We verify the node is on the latest cluster state version, and manages no buckets, so that should be sufficient?  Let me remove the check on all nodes being up.